app/code/Magento/Catalog/Model/ResourceModel/Category/Collection.php

```diff
@@ -556,8 +524,7 @@ private function getProductsCountFromCategoryTable(Category $item, string $websi
      */
     private function getProductsCountQuery(array $categoryIds, $addVisibilityFilter = true): Select
     {
-        $connections = $this->_resource->getConnection();
-        $categoryTable = $connections->getTableName('catalog_category_product_index');
+        $categoryTable = $this->tableMaintainer->getMainTable($this->getProductStoreId());
```

Comment:
Given that these tables are dynamic, in the event the index table for the store does not exist, we should just return an associative array of `$categoryId => 0`